hooks/browser_integration.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import sgtk
+
+class BrowserIntegration(sgtk.Hook):
+    def get_cache_key(self, config_uri, project, entity_type):
```

Comment:
We can discuss this later but curious about the right level of info here - would it make sense to pass down more state here that a hook could take advantage of? What would be pros/cons be? Let's disucss!